### ISSUE TITLE
Zed automatically downloads and runs pylsp

### ISSUE THREAD

**Issue by injust** [zed-industries/zed#22471]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Zed automatically downloads and runs pylsp. This was apparently added in https://github.com/zed-industries/zed/pull/20358, but is not documented anywhere except for a single line in the release notes.

https://zed.dev/docs/languages/python still says that pyright is the only language server for Python.

This is not helped by https://github.com/zed-industries/zed/issues/22472. "Why is this running?" is not a question that users should be asking.

### Zed Version and System Specs

Zed: v0.166.1 (Zed)
OS: macOS 15.2.0
Memory: 16 GiB
Architecture: x86_64

**Comment by senpos** [comment]
I encountered this issue today as a new user revisiting Zed after hearing it mentioned on a podcast.  

Upon opening a Python project, I noticed `pylsp` unexpectedly appearing in the bottom left corner. As a result, I started receiving warnings from `mccabe` and possibly other tools bundled with `pylsp`.  

This behavior is concerning for a few reasons:  
1. The editor appears to install an extension (`pylsp`) that is not documented in the [official Python support page](https://zed.dev/docs/languages/python), which explicitly states that only `tree-sitter-python` and `microsoft/pyright` are installed, with no formatting or linting tools included by default.  
2. The `pylsp` extension is not listed on the "Extensions" page.  
3. I checked the [auto-install extensions documentation](https://zed.dev/docs/configuring-zed#auto-install-extensions) and inspected the directory where such extensions are supposed to be stored (`~/Library/Application Support/Zed/extensions/installed`). However, `pylsp` is not located there either.

**Comment by JosephTLyons** [comment]
Yeah, it seems we have some outdated docs. Also, I'm guessing this is the location of the language server:

`~/Library/Application Support/Zed/languages/pylsp`

**Comment by bulletmark** [comment]
I was wondering why `pyflakes` warnings were being presented to me because the documentation states the only Python lsp/linter run is `pyright`. Certainly the documentation needs to be updated as reported here but there is another related issue that the default Python LSP etc settings can not be viewed. Doing "Open Default Settings" shows nothing about these settings. That is as unnerving as the documentation being out of date.

**Comment by injust** [comment]
> but there is another related issue that the default Python LSP etc settings can not be viewed. Doing "Open Default Settings" shows nothing about these settings. That is as unnerving as the documentation being out of date.

https://github.com/zed-industries/zed/issues/22472 covers that (although I'm waiting for it to be reopened).

**Comment by injust** [comment]
Fixed by https://github.com/zed-industries/zed/pull/26835.

Also see https://github.com/zed-industries/zed/issues/26820#issuecomment-2726719982 for context on why Python LSP settings aren't included in the default settings.
